contracts/carrot-app/src/msg.rs

```diff
@@ -13,52 +14,52 @@ abstract_app::app_msg_types!(App, AppExecuteMsg, AppQueryMsg);
 /// App instantiate message
 #[cosmwasm_schema::cw_serde]
 pub struct AppInstantiateMsg {
-    /// Id of the pool used to get rewards
-    pub pool_id: u64,
-    /// Seconds to wait before autocompound is incentivized.
-    pub autocompound_cooldown_seconds: Uint64,
-    /// Configuration of rewards to the address who helped to execute autocompound
-    pub autocompound_rewards_config: AutocompoundRewardsConfig,
+    /// Strategy to use to dispatch the deposited funds
+    pub balance_strategy: BalanceStrategy,
+    /// Configuration of the aut-compounding procedure
+    pub autocompound_config: AutocompoundConfig,
+    /// Target dex to swap things on
+    pub dex: String,
     /// Create position with instantiation.
     /// Will not create position if omitted
-    pub create_position: Option<CreatePositionMessage>,
-}
-
-#[cosmwasm_schema::cw_serde]
-pub struct CreatePositionMessage {
-    pub lower_tick: i64,
-    pub upper_tick: i64,
-    // Funds to use to deposit on the account
-    pub funds: Vec<Coin>,
-    /// The two next fields indicate the token0/token1 ratio we want to deposit inside the current ticks
-    pub asset0: Coin,
-    pub asset1: Coin,
-    // Slippage
-    pub max_spread: Option<Decimal>,
-    pub belief_price0: Option<Decimal>,
-    pub belief_price1: Option<Decimal>,
+    pub deposit: Option<Vec<Coin>>,
 }
 
 /// App execute messages
 #[cosmwasm_schema::cw_serde]
 #[cfg_attr(feature = "interface", derive(cw_orch::ExecuteFns))]
 #[cfg_attr(feature = "interface", impl_into(ExecuteMsg))]
 pub enum AppExecuteMsg {
-    /// Create the initial liquidity position
-    CreatePosition(CreatePositionMessage),
     /// Deposit funds onto the app
-    Deposit {
-        funds: Vec<Coin>,
-        max_spread: Option<Decimal>,
-        belief_price0: Option<Decimal>,
-        belief_price1: Option<Decimal>,
-    },
+    /// Those funds will be distributed between yield sources according to the current strategy
+    /// TODO : for now only send stable coins that have the same value as USD
+    /// More tokens can be included when the oracle adapter is live
+    Deposit { funds: Vec<Coin> },
     /// Partial withdraw of the funds available on the app
-    Withdraw { amount: Uint128 },
-    /// Withdraw everything that is on the app
-    WithdrawAll {},
+    /// If amount is omitted, withdraws everything that is on the app
+    Withdraw { amount: Option<Uint128> },
     /// Auto-compounds the pool rewards into the pool
     Autocompound {},
+    /// Rebalances all investments according to a new balance strategy
+    Rebalance { strategy: BalanceStrategy },
```

Comment:
I'd think `Rebalance` is an action similar to `Autocompound` but it also rebalances funds. 

Maybe this can be renamed to `UpdateStrategy`?

Reply:
Yes ! What's thr target behavior in this endpoint. It could just update strategy, it could also update strategy and rebalance the funds, what di you think ?

Reply:
Update and rebalance, but imo the rebalance is implied when you update it. Would be weird to have to call it twice!